[DOCS] CTI enhancements to the Overview page and Alert details pane (#808)
elastic/security-docs: docs/detections/alerts-ui-manage.asciidoc

[role="xpack"]
== Manage detection alerts

The Alerts page displays all <<detection-alert-def, detection alerts>>.
From the Alerts table, you can filter alerts, change an alert's status, and start
investigating and analyzing alerts in Timeline.

TIP: From *Timelines*, you can <<cases-ui-open, create cases>> to track issues and
share information with colleagues.

[float]
[[detection-view-and-filter-alerts]]
=== View and filter detection alerts
The Alerts page offers various ways for you to organize and triage detection alerts as you investigate suspicious events. You can:

* Filter for a specific rule in the KQL bar (for example,
`signal.rule.name :"SSH (Secure Shell) from the Internet"`).

NOTE: KQL autocomplete for `.siem-signals-*` indices is available on the
*Alerts* and *Rule details* pages, and in Timeline when either `All` or
`Detection alerts` is selected.

* Use the date and time filter to select a time range that you’re interested in exploring. By default, this filter is set to search the last 24 hours.

> * View detection alerts generated by a specific rule. To do this, click *Detect* -> *Alerts* -> *Manage detection rules*, then click on a rule name in the All rules table. The *Rules detail page* displays a comprehensive view of the rule's details, and alert details are displayed in the Alerts table beneath the Trend histogram.

Reviewer: If you're on the Alerts page, the `Manage detection rules` link is still there; alternately if you're browsing via navigation sidebar the `Rules` link is a more direct route.

[role="xpack"]
== Manage detection alerts

The Alerts page displays all <<detection-alert-def, detection alerts>>.
From the Alerts table, you can filter alerts, change an alert's status, and start
investigating and analyzing alerts in Timeline.

TIP: From *Timelines*, you can <<cases-ui-open, create cases>> to track issues and
share information with colleagues.

[float]
[[detection-view-and-filter-alerts]]
=== View and filter detection alerts
The Alerts page offers various ways for you to organize and triage detection alerts as you investigate suspicious events. You can:

* Filter for a specific rule in the KQL bar (for example,
`signal.rule.name :"SSH (Secure Shell) from the Internet"`).

NOTE: KQL autocomplete for `.siem-signals-*` indices is available on the
*Alerts* and *Rule details* pages, and in Timeline when either `All` or
`Detection alerts` is selected.

* Use the date and time filter to define a specific time range. By default, this filter is set to search the last 24 hours.

> * To view detection alerts generated by a specific rule. To do this, go to *Detect* -> *Rules*, and then select a rule name in the All rules table. The *Rules detail page* displays a comprehensive view of the rule's details, and alert details are displayed in the Alerts table beneath the Trend histogram. Alternatively, you can go to *Detect* -> *Alerts* -> *Manage detection rules*, and then click on a rule name in the All rules table to view the alerts associated with the rule.

Reviewer 2: I kind of liked the original beginning of this line ("View detection alerts" instead of "To view detection alerts"), because it matches the form of the other bullets, each starting with an active verb and flowing from the "You can..." lead-in to the list (line 15).

* Use the *Stack by* drop-down in the Trend histogram to select specific parameters to visualize the individual counts. For example, if you select `signal.rule.name`, the histogram displays the total counts by alert name.

> * To filter alert results to include building block alerts or to only show alerts from indicator match rules, select the *Additional filters* drop-down. By default, building block alerts are excluded from the Alerts table; therefore, including them expands the number of alerts.

Reviewer 2: Similar issue with the verb at the beginning of the line. Possible revision:
```suggestion
* Filter alert results to include building block alerts or to only show alerts from indicator match rules. To do this, select the *Additional filters* drop-down. By default, building block alerts are excluded from the Alerts table; therefore, including them expands the number of alerts.
```

> NOTE: When updating alert results to include building block alerts, the Security app searches the `.siem-signals-<Kibana space>` index for the `signal.rule.building_block_type` field. When looking for alerts created from indicator match rules, the app searches the same index for the `signal.rule.threat_mapping` field.

Reviewer 2: Revise to "Security Solution" or maybe "Elastic Security"?
```suggestion
NOTE: When updating alert results to include building block alerts, the Security Solution searches the `.siem-signals-<Kibana space>` index for the `signal.rule.building_block_type` field. When looking for alerts created from indicator match rules, the Security Solution searches the same index for the `signal.rule.threat_mapping` field.
```

Author: Keeping for now since we use "app" elsewhere in our docs. :)

- `file.elf.telfhash`: Imports in an ELF file
- `file.hash.ssdeep`: The SSDEEP hash
- `source.ip`: The IP address of the source (IPv4 or IPv6)
- `destination.ip`*: The event's destination IP address
- `url.full`: The full URL of the event source
- `registry.path`: The full registry path, including the hive, key, and value

Additional threat intelligence data is generated for indicator match alerts when indicators are matched with source events, as outlined by the conditions in the relevant indicator match rule.

*Threat Intel tab*

Shows the number of individual threats matching the alert in reverse chronological order, with the most recent threats at the top. Each threat is labelled with values from its `matched.field` and `matched.atomic` fields. All mapped fields are displayed for each matched threat.

NOTE: If the alert does not contain threat intelligence data, the *Threat Intel* tab displays a message that none is available and provides a link to Threat Intel module documentation.

*Table tab*

Shows the alert details in table format. Alert details are organized into field value pairs.

*JSON View tab*

Shows the alert details in JSON format.

> [role="screenshot"]

Author: Hm, I'm not seeing this on my local version but will keep an eye out for this in the next draft.

[float]
[[view-alert-details]]
=== View alert details

> To further inspect an alert, click the *View details* button from the Alerts table. The Alert details flyout appears and offers you several ways to view alert data.

Reviewer 3: ```suggestion
To further inspect an alert, click the *View details* button from the Alerts table. The Alert details flyout appears and provides several options to view alert data.
```

[role="screenshot"]
image::images/view-alert-details.png[Shows the Event Renderer button, 200]

Each of the tabs within the Alert details flyout are described below.

[discrete]
*Summary tab*

> The *Summary* tab provides an overview of the alert and shows relevant threat intelligence details. You can use this information to gain an understanding of what caused the alert and to make decisions about handling it.

Reviewer 3: Open to alternate wording here -- just trying to keep it succinct! 
```suggestion
The *Summary* tab provides an overview of the alert and shows relevant threat intelligence details. Use this information to analyze what generated the alert, so you can triage its relevance and resolve it as appropriate.
```

The *Summary* tab always begins with a list of general alert details. These details are conveyed as a curated list of fields containing essential and available information about the alert. Within this section, you can find the alert's current status, locate the rule that produced the alert, learn the alert's severity, and more. You can use these details to start forming a sense of what caused the alert and then continue your investigation by checking out available threat intelligence on the alert.

> The {es-sec} app collects threat intelligence data for all alerts. Fields that show threat intelligence data are appended with an informational icon and placed after general alert details on the *Summary* tab.

Reviewer 3: When referring to the UI, `{security-app}` is OK to use. However, when describing function, we'll want to (generally) use `{elastic-sec}`. There are times where we'll have to even be more specific, e.g. `Elastic Endpoint`. 
```suggestion
{elastic-sec} collects threat intelligence data for all alerts. Fields that show threat intelligence data are appended with an informational icon and placed after general alert details on the *Summary* tab.
```

> When gathering threat intelligence, the {es-sec} app  queries indicator data from the past 30 days to generate enrichments for the below fields.

Reviewer: I would avoid the verb `generate` as that could be interpreted as generating _persistent_ data, which this is not:

```suggestion
When gathering threat intelligence, the {es-sec} app queries indicator data from the past 30 days to find enrichments for the below fields.
```

Reviewer: I would also avoid the `enrichments` term for now:
```suggestion
When gathering threat intelligence, the {es-sec} app queries indicator data from the past 30 days to find threat intelligence for the below fields.
```

> If these field values aren't found, the {es-sec} app does not create threat intelligence for the alert.

Reviewer: Similar to above:
```suggestion
If these fields aren't present, the {es-sec} app does not perform the query and no additional threat intelligence is displayed.
```

NOTE: You can learn more about the query by going to the **Threat Intel** tab and clicking the **Inspect** icon next to threat you want to inspect.

- `file.hash.md5`: The MD5 hash
- `file.hash.sha1`: The SHA1 hash
- `file.hash.sha256`: The SHA256 hash
- `file.pe.imphash`: Imports in a PE file
- `file.elf.telfhash`: Imports in an ELF file
- `file.hash.ssdeep`: The SSDEEP hash
- `source.ip`: The IP address of the source (IPv4 or IPv6)
- `destination.ip`*: The event's destination IP address
- `url.full`: The full URL of the event source
- `registry.path`: The full registry path, including the hive, key, and value

> Additional threat intelligence is generated for indicator match alerts. These details are produced when indicators are matched with source events, as outlined by the conditions in the relevant indicator match rule. Any data that's duplicated is removed.

Reviewer: I don't think the note about duplicated data is necessary: if we did not do this and instead displayed the same indicator twice, most users would consider that a bug. 

This phrase could also potentially be misinterpreted as: "If you have duplicated indicators in your index, we will de-dup them for you," which is not true.

```suggestion
Additional threat intelligence is generated for indicator match alerts. These details are produced when indicators are matched with source events, as outlined by the conditions in the relevant indicator match rule.
```

*Threat Intel tab*

> Shows the number of threats linked the alert and displays them individually. Threat titles are formed using values from the `matched.field` and `matched.atomic` fields and threats appear in reverse chronological order, with the most recent at the top. Available `threat.indicator` and `source.event` data is displayed for each threat.

Reviewer 3: ```suggestion
Shows the number of individual threats linked to the alert in reverse chronological order, with the most recent threats at the top. Threat titles are formed using values from the `matched.field` and `matched.atomic` fields. Available `threat.indicator` and `source.event` data is displayed for each threat.
```
Question -- what do you mean by "available `threat.indicator` data is displayed? Is it always there? If so, take out `available`.

Reviewer: A few notes here: 

* "linked" is a verb we haven't used previously, I would stick with "matched"
* It might not be clear what a "threat title" is, I would just refer to that heading as a "label"
* I'm skeptical of the utility of that last sentence as that's more of an implementation detail subject to change, and the available fields are self-evident. _Maybe_ you could say "All mapped fields are displayed for each matched threat/indicator." if you want to qualify it.

```suggestion
Shows the number of threats matching the alert in reverse chronological order, with the most recent at the top. Each threat is labelled with values from its `matched.field` and `matched.atomic` fields. All mapped fields are displayed for each matched threat.
```

> If the alert does not contain threat intelligence data, the *Threat Intel* tab displays a message indicating that none is available and provides a link to Threat Intel module documentation.

Reviewer 3: ```suggestion
NOTE: If the alert does not contain threat intelligence data, the *Threat Intel* tab displays a message that none is available and provides a link to Threat Intel module documentation.
```

*Table tab*

Shows the alert details in table format. Alert details are organized into field value pairs.

*JSON View tab*

Shows the alert details in JSON format.

> // * *Summary*: Shows an aggregated view of alert details, plus two types of threat intelligence data: indicator rule enrichments and investigation time enrichments.

Reviewer: I think we can just delete these comments for now 👍; we can always resurrect them from git if we later need them.

The Alert details flyout appears and offers several options for viewing alert details:

> * *Summary*: Shows an aggregated view of alert details, plus two types of threat intelligence data: indicator rule enrichments and investigation time enrichments.

Reviewer 4: I remember @shimonmodi mentioning that we shouldn't use these phrases (indicator match enrichments and investigation time enrichments), but I am not entirely sure exactly what we agreed on instead - we might want to check with @shimonmodi and @devonakerr.

Reviewer: For context, here is the latest copy for those pages: https://github.com/elastic/kibana/pull/105701

> ** *Indicator rule enrichments*: As the name suggests, indicator rule enrichments only display for alerts produced by indicator match rules. You can spot an indicator rule enrichment by checking for fields pre-fixed by the {es-sec} icon and appended with source event data, if that information is available. Indicator rule enrichments are created when rule indicators are met and mapped index fields are produced. An alert can have one or more indicator rule enrichment and the same indicator match can display multiple times.

Reviewer 4: in the first sentence we could reconsider the word `display` as it sounds like they could have existed somewhere else but `display` only for indicator match rule alerts, when in reality they don't exist otherwise. In the second sentence it sounds like we would have displayed the icon anyway and `append source event data` only if it exists, instead we could just say something along the lines of `when there is a threat match you can see the matched.field and matched atomic next to the {es-sec}  icon`. The next sentence sounds a bit awkward, we could say something like `Indicator rule enrichments are created when threat indicators are matched with source events according to the conditions outlined in the relevant indicator match rule definition`. From the next sentence we can remove `and the same indicator match can display multiple times` as they are not supposed to be the same match, just looks the same in the summary area, we have a ticket to improve the UX around that.

> ** *Indicator rule enrichments*: As the name suggests, indicator rule enrichments only display for alerts produced by indicator match rules. You can spot an indicator rule enrichment by checking for fields pre-fixed by the {es-sec} icon and appended with source event data, if that information is available. Indicator rule enrichments are created when rule indicators are met and mapped index fields are produced. An alert can have one or more indicator rule enrichment and the same indicator match can display multiple times.

Reviewer: This has also changed; we now distinguish these matches with an info icon appended after the field name (ref: https://github.com/elastic/kibana/pull/105701)

> ** *Indicator rule enrichments*: As the name suggests, indicator rule enrichments only display for alerts produced by indicator match rules. You can spot an indicator rule enrichment by checking for fields pre-fixed by the {es-sec} icon and appended with source event data, if that information is available. Indicator rule enrichments are created when rule indicators are met and mapped index fields are produced. An alert can have one or more indicator rule enrichment and the same indicator match can display multiple times.

Reviewer: Also of note: these fields will be displayed after any other summary fields.

> ** *Investigation time enrichments*: Displays for all alerts regardless of the rule type that generated it. Like indicator rule enrichments, investigation time enrichments are also pre-fixed by the {es-sec} icon as a way to set them apart from general alert details. Time enrichment data from the past 30 days is generated when the {es-sec} app queries the `filebeat-*` index for the following set of fields. If these field values aren't found, investigation time enrichments aren't created. Or, if an indicator match rule uses these fields for indicator mapping, and the indicator rule enrichment data matches the  investigation time enrichment data, the investigation time enrichments are treated as duplicated data and are removed. For example, if you specify `file.hash.sha256` and `file.hash.md5` as fields that should be mapped for your indicator match rule, these fields will only display as indicator rule enrichments on the **Summary** and **Threat Intel** tabs when the rule is activated and the alert is generated. You can learn more about the investigation time enrichment query by going to the **Threat Intel** tab and clicking the **Inspect** icon next to event you want to inspect.

Reviewer 2: Again, omit hyphen in "prefixed." 

Also, is it possible to move the list of values up, to immediately follow "the following set of fields" in sentence 3? That might also help break up the very long paragraph.

> ** *Investigation time enrichments*: Displays for all alerts regardless of the rule type that generated it. Like indicator rule enrichments, investigation time enrichments are also pre-fixed by the {es-sec} icon as a way to set them apart from general alert details. Time enrichment data from the past 30 days is generated when the {es-sec} app queries the `filebeat-*` index for the following set of fields. If these field values aren't found, investigation time enrichments aren't created. Or, if an indicator match rule uses these fields for indicator mapping, and the indicator rule enrichment data matches the  investigation time enrichment data, the investigation time enrichments are treated as duplicated data and are removed. For example, if you specify `file.hash.sha256` and `file.hash.md5` as fields that should be mapped for your indicator match rule, these fields will only display as indicator rule enrichments on the **Summary** and **Threat Intel** tabs when the rule is activated and the alert is generated. You can learn more about the investigation time enrichment query by going to the **Threat Intel** tab and clicking the **Inspect** icon next to event you want to inspect.

Reviewer 4: We could consider replacing the sentence `Time enrichment data from the past 30 days is generated when the {es-sec} app queries the filebeat-* index for the following set of fields` with something like `Indicator data from the past 30 days is queried to generate enrichments for the fields listed below.` and we can skip the next sentence. 

We can replace the sentence `Or, if an indicator match rule uses these fields for indicator mapping, and the indicator rule enrichment data matches the  investigation time enrichment data, the investigation time enrichments are treated as duplicated data and are removed` with something like `if investigation time matches return enrichments that are identical to indicator rule matches they will not appear twice`. 

In the next sentence we can say `are mapped` instead of `should be mapped`, and we can also say `when alerts are generated` to finish that sentence, as the rule can generate many alerts.

- `file.hash.md5`: The MD5 hash
- `file.hash.sha1`: The SHA1 hash
- `file.hash.sha256`: The SHA256 hash
- `file.pe.imphash`: Imports in a PE file
- `file.elf.telfhash`: Imports in an ELF file
- `file.hash.ssdeep`: The SSDEEP hash
- `source.ip`: The IP address of the source (IPv4 or IPv6)
- `destination.ip`*: The event's destination IP address
- `url.full`: The full URL of the event source
- `registry.path`: The full registry path, including the hive, key, and value

> NOTE: To further verify the type of threat intelligence data you're viewing, hover your mouse over the field name. Also note that if an alert is linked to more than one threat, matched field values  display within the *Summary* tab and a more detailed view can be found within the *Threat Intel* tab.

Reviewer 4: maybe we can just say `hover over the field name` as that's the only way to hover on something as far as I understand. Although I think for accessibility users should be able to pop that up with the keyboard as well.

> * *Threat Intel*: Shows the number of threats linked the alert and displays indicator rule enrichments and investigation time enrichments individually. Threats appear in reverse chronological order, with the most recent events at the top. Threat titles are formed using matched fields and atomic indicators, and available `threat.indicator` and `source.event` data is displayed for each threat. If the alert does not contain indicator rule enrichments or investigation time enrichments, the *Threat Intel* tab displays the message "No Indicator Match or Threat Intel Enrichment Found" and provides a link to Threat Intel module documentation. Similarly, if the alert does not contain indicator rule enrichments, the *Threat Intel* tab displays the message "No indicator match rule enrichments found" and provides a link to Threat Intel module documentation.

Reviewer 2: Missing word. Also, would it help to include the link to Threat Intel docs here, too (mentioned in last sentence)?
```suggestion
* *Threat Intel*: Shows the number of threats linked to the alert and displays indicator rule enrichments and investigation time enrichments individually. Threats appear in reverse chronological order, with the most recent events at the top. Threat titles are formed using matched fields and atomic indicators, and available `threat.indicator` and `source.event` data is displayed for each threat. If the alert does not contain indicator rule enrichments or investigation time enrichments, the *Threat Intel* tab displays the message "No Indicator Match or Threat Intel Enrichment Found" and provides a link to Threat Intel module documentation. Similarly, if the alert does not contain indicator rule enrichments, the *Threat Intel* tab displays the message "No indicator match rule enrichments found" and provides a link to Threat Intel module documentation.
```

Reviewer: The copy "No Indicator Match or Threat Intel Enrichment Found" is outdated and now could be one of three states (ref: https://github.com/elastic/kibana/pull/105701); it might be sufficient to mention "a message is displayed" instead of referencing specific copy.

> * *Threat Intel*: Shows the number of threats linked the alert and displays indicator rule enrichments and investigation time enrichments individually. Threats appear in reverse chronological order, with the most recent events at the top. Threat titles are formed using matched fields and atomic indicators, and available `threat.indicator` and `source.event` data is displayed for each threat. If the alert does not contain indicator rule enrichments or investigation time enrichments, the *Threat Intel* tab displays the message "No Indicator Match or Threat Intel Enrichment Found" and provides a link to Threat Intel module documentation. Similarly, if the alert does not contain indicator rule enrichments, the *Threat Intel* tab displays the message "No indicator match rule enrichments found" and provides a link to Threat Intel module documentation.

Reviewer 4: instead of `Threat titles are formed using matched fields and atomic indicators` we could probably say something like `Enrichment titles are formed using matched.field and matched.atomic values`
